File: C-5-SDT/OOP_and_Python_Programming/M2.5/I_Smallest_Pair.py
# https://codeforces.com/group/MWSDmqGsZm/contest/219774/problem/I
tc = int(input())

while tc > 0:
    tc -= 1
    
    n = int(input())
    
    # 1-based indexing (same as your C++ code)
    arr = [0] * (n + 1)
    
    values = list(map(int, input().split()))
    for i in range(1, n + 1):
        arr[i] = values[i - 1]
    
    # find smallest result
    # small_sum = float('inf')
    small_sum = 10**9
    
    for i in range(1, n):
        for j in range(i + 1, n + 1):
            cal = arr[i] + arr[j] + (j - i)
            small_sum = min(small_sum, cal)
    
    print(small_sum)

File: C-2-CPP/module3.5/I_Smallest_Pair.cpp
// https://codeforces.com/group/MWSDmqGsZm/contest/219774/problem/I

#include <iostream>
using namespace std;
int main()
{
    int tc;
    cin >> tc;
    while (tc--)
    {
        int n;
        cin >> n;

        // full work with 1 index based
        int arr[n + 1];
        for (int i = 1; i <= n; i++)
        {
            // scanf("%d", &arr[i]);
            cin >> arr[i];
        }

        // method 2: find possible smallest result 1 - N
        int small_sum = INT_MAX;
        for (int i = 1; i <= n - 1; i++)
        {
            for (int j = i + 1; j <= n; j++)
            {
                int cal = arr[i] + arr[j] + j - i;
                // if (cal < small_sum)
                // {
                //     small_sum = cal;
                // }
                small_sum = min(cal, small_sum);
            }
        }

        // printf("%d\n", small_sum);
        cout << small_sum << endl;
    }

    return 0;
}
